crates/router/src/core/unified_connector_service/transformers.rs

```diff
@@ -994,3 +997,106 @@ impl ForeignTryFrom<common_types::payments::CustomerAcceptance>
         })
     }
 }
+
+// =================== WEBHOOK TRANSFORMERS ===================
+
+/// Transform webhook request details into UCS gRPC format
+impl ForeignTryFrom<&hyperswitch_interfaces::webhooks::IncomingWebhookRequestDetails<'_>>
+    for payments_grpc::RequestDetails
+{
+    type Error = error_stack::Report<UnifiedConnectorServiceError>;
+
+    fn foreign_try_from(
+        request_details: &hyperswitch_interfaces::webhooks::IncomingWebhookRequestDetails<'_>,
+    ) -> Result<Self, Self::Error> {
+        let headers_map = request_details
+            .headers
+            .iter()
+            .map(|(key, value)| {
+                let value_string = value.to_str().unwrap_or_default().to_string();
+                (key.as_str().to_string(), value_string)
+            })
+            .collect();
+
+        Ok(Self {
+            method: 1, // POST method for webhooks
+            uri: Some(
+                request_details
+                    .headers
+                    .get("x-forwarded-path")
+                    .and_then(|h| h.to_str().ok())
+                    .unwrap_or("/webhook")
```

Comment:
we shouldn't hardcode as webhook, instead add "unknown" or related to that which say the header data is missing